subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c

```diff
@@ -64,13 +65,14 @@ static inline CborError
 os_mgmt_taskstat_encode_thread_name(struct CborEncoder *encoder, int idx,
 				    const struct k_thread *thread)
 {
-	int err = 0;
+	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+	const int max_size = MIN(ARRAY_SIZE(thread_name), ARRAY_SIZE(thread->name)) - 1;
 
 	ARG_UNUSED(idx);
+	memcpy(thread_name, thread->name, max_size);
+	thread_name[max_size] = '\0';
```

Comment:
why not just do:
```
size_t name_length = strlen(thread->name);
if (name_length > CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1) {
    name_length = CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1;
}
err = cbor_encode_text_string(encoder, thread->name, name_length);
```

what's the reason for copying to a new array ?

Reply:
> 
> what's the reason for copying to a new array ?

Hm. Yes, that wasn't the smartest way to do this.